pkg/apis/kubeone/v1beta1/types.go

```diff
@@ -0,0 +1,321 @@
+/*
+Copyright 2020 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KubeOneCluster is KubeOne Cluster API Schema
+type KubeOneCluster struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Name is the name of the cluster
+	Name string `json:"name"`
+	// ControlPlane describes the control plane nodes and how to access them
+	ControlPlane []HostConfig `json:"controlPlane,omitempty"`
```

Comment:
@xmudrii how do you think, does it make sense to turn this into a struct with inside .Hosts field?

Reply:
Do you mean something along:
```go
type Hosts struct {
    ControlPlane  []HostConfig `json:"controlPlane"`
    StaticWorkers []HostConfig `json:"staticWorkers"`
}
```

In this case `StaticWorkers` can even be called just `Workers`, but I'd maybe prefer `StaticWorkers`.

Did you think of something like this?

Reply:
not quite, more like 
```go
...
ControlPlane ControlPlane `json:"controlPlane"`
...
type ControlPlane struct {
    Hosts  []HostConfig `json:"hosts,omitempty"`
}
```

Reply:
Adding something to `type ControlPlane struct` at later times is easy, unlike to `[]Hosts`.

Reply:
@kron4eg Updated, PTAL.